src/main/java/com/google/devtools/build/lib/bazel/BazelRepositoryModule.java

```diff
@@ -360,6 +360,11 @@ public void beforeCommand(CommandEnvironment env) throws AbruptExitException {
       }
 
       if (repoOptions.repositoryOverrides != null) {
+        if (pkgOptions != null && !pkgOptions.checkExternalFiles) {
+          env.getReporter()
+              .handle(Event.warn("--noexperimental_check_external_files is making it so we don't check the disk for changes to your repositories overriden with --override_repository"));
+        }
```

Comment:
nit: this is phrased a little strangely to my eye. I might say instead:

"Changes to external repositories overridden by --override_repository will not be detected because --noexperimental_check_external_files is set".